docs/maintenance/index.md

```diff
@@ -0,0 +1,13 @@
+# Maintenance
+
+```{toctree}
+:hidden:
+
+
+release_process
+```
+
+Documentation related to the maintenance of this project. In other words, docs (mostly)
+for maintainers by maintainers. The areas covered are the following:
```

Comment:
I think this can just go into the contributing section; releasing is a form of contributing, and I'd like to keep the number of top-level sections low.

Reply:
Fair enough, the maintenance section was my original plan but putting this in contributing works too.

Reply:
I just put it where @ichard26 said to.